src/emqx_client.erl

```diff
@@ -135,23 +154,47 @@
 
 -type(subscribe_ret() :: {ok, properties(), [reason_code()]} | {error, term()}).
 
--export_type([client/0, topic/0, qos/0, properties/0, payload/0,
-              packet_id/0, pubopt/0, subopt/0, reason_code/0]).
+-type(request_input() :: binary()).
 
-%% Default timeout
--define(DEFAULT_KEEPALIVE,       60000).
--define(DEFAULT_ACK_TIMEOUT,     30000).
--define(DEFAULT_CONNECT_TIMEOUT, 60000).
+-type(response_payload() :: binary()).
 
--define(PROPERTY(Name, Val), #state{properties = #{Name := Val}}).
+-type(request_handler() :: fun((request_input()) -> response_payload())).
+
+-type(response_topic() :: ?SHARED(ShareGroup :: binary(), binary())).
+
+-export_type([client/0, topic/0, qos/0, properties/0, payload/0,
```

Comment:
`export_type` is from before this change.
but it seems wrong to export `qos/0`, `topic/0`, `packet_id/0` (maybe more) from this module.
these types seem to be common enough to be either defined in and exported from emqx.erl.
topic/0 is defined as `emqx_topic:topic()` which is fair, but worth to expose it from emqx.erl too.
i.e. in emqx.erl:   `-type topic() :: emqx_topic:topic()`.

Also, some of the types (e.g. `response_payload`) is used internally only, so there is no need to export.